centreon/www/api/class/centreon_ceip.class.php

```diff
@@ -44,18 +44,33 @@ public function __construct()
     {
         parent::__construct();
 
+        /** @var \Centreon $centreon */
         global $centreon;
 
         $this->user = $centreon->user;
 
+        if (null === $this->pearDB) {
+            throw new LogicException('No DB connector given.');
+        }
         // Generate UUID
         $this->uuid = (string) (new CentreonUUID($this->pearDB))->getUUID();
 
         $kernel = \App\Kernel::createForWeb();
-        $this->logger = $kernel->getContainer()->get(\Centreon\Domain\Log\Logger::class)
-            ?? throw new LogicException('Logger not found in container');
-        $this->featureFlags = $kernel->getContainer()->get(\Core\Common\Infrastructure\FeatureFlags::class)
-            ?? throw new LogicException('FeatureFlags not found in container');
+
+        /** @var null|\Centreon\Domain\Log\Logger $logger */
+        $logger = $kernel->getContainer()->get(\Centreon\Domain\Log\Logger::class);
+        if (null === $logger) {
+            throw new LogicException('Logger not found in container');
+        }
+
+        /** @var null|\Core\Common\Infrastructure\FeatureFlags $featureFlags */
+        $featureFlags = $kernel->getContainer()->get(\Core\Common\Infrastructure\FeatureFlags::class);
+        if (null === $featureFlags) {
+            throw new LogicException('FeatureFlags not found in container');
+        }
+
+        $this->logger = $logger;
+        $this->featureFlags = $featureFlags;
```

Comment:
Why did you replace this code when it was already working properly?
Please undo your changes.